### PR TITLE
Nm 130 voxel

### PR DESCRIPTION
**Pull request by ravisoni31**

### CODE REVIEW

**Comment by ravisoni31** [comment]
The validation video links will be added tomorrow.

**Comment by void-robotics** [comment]
looks pretty good @ravisoni31; i would also
1. update single letter variables
2. not update the bird object, as i simply need to see shapes rotating; and the bird object is the next task
3. prove that they are equally spaced; not sure how though

**Comment by ravisoni31** [comment]
Jenkins build #: 23 

1. Most of the single letter variables have been updated except (x,y,z) and (a,b,c) which are standard for ellipsoid equations.
2. Sure. It has been taken care of.
3. I tried to rotate the parts by adding bird_part.rotate(..) in Bird.cpp; it throws an error: `"didn't call Part::setRotatingPoint or the rotating point was replaced with the same Point values without the boolean...exiting"` which is coming from the "Getters and Setters" part in Part.cpp file.  Or am I doing something wrong here?

**Comment by void-robotics** [comment]
Generally for Newmind error messages, try to find the source of the error as it helps in debugging. Basically, Part::rotate requires one point in the Part to have been specified to be the center of rotation using Part::setRotatingPoint; otherwise, Part::rotate won't know where to rotate about.

to search all files for a regex, try the linux command:
grep -rn 'root_dir' -e "regex"

File: src/simulation/nodes/SimulationTester.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    ros::init(argc, argv, "simulator_test");
+    ros::NodeHandle node;
+    GenUtil gen_util;
+    signal(SIGINT, shutdownSimNode);
+
+	int simulator_shape = 1;
+	if(! ros::param::get("/simulation/shape", simulator_shape)) ROS_ERROR("failed to get param");
+
+    switch(simulator_shape) {
+    case 0: { // Ellipsoid
```
Comment (wildomonges) [inline src/simulation/nodes/SimulationTester.cpp]: I suggest to use constants instead magic numbers. For example, you can define at the top of the source code
const int ELLIPSOID = 0;
const int PRISM = 1;
const int AIRFOIL = 2;
and use those constants inside the switch like
case ELLIPSOID: { //do anything}
case PRISM: {//do anything}
I think the code will be more readable and understandable.

Comment (void-robotics) [reply src/simulation/nodes/SimulationTester.cpp]: Normally I'd agree @wildomonges, but this code is in a "Tester" executable, meaning it's only for short tests that will eventually be deleted. These are not unit tests; instead, they are temporary system tests.

**Comment by ravisoni31** [comment]
Jenkins build no.: 36 (FAILED)

> The newmind repository is successfully built!
> 
> *********************************************************************
> Building Tests
> *********************************************************************
> Scanning dependencies of target tests
> Built target tests
> 
> *********************************************************************
> Running Tests
> *********************************************************************
> Running tests...
> Test project /var/lib/jenkins/jenkins-build-origin/nm-130-voxel/build
> No tests were found!!!
> ERROR: Unable to communicate with master!
> Build step 'Execute shell' marked build as failure
> Finished: FAILURE

**Comment by void-robotics** [comment]
sorry; not sure how that happened; but the roscore is up now

**Comment by ravisoni31** [comment]
Jenkins build no: 37 (SUCCESS)

**Review by ghost — APPROVED** [review]

**Pull request merged**